File: 品味Java的21種設計模式/dp_java/factorymethod/example7/ExportDB.java
package cn.javass.dp.factorymethod.example7;
/**
 * 导出成数据库备份文件形式的对象
 */
public class ExportDB implements ExportFileApi {
    public boolean export(String data) {
        //简单示意一下，这里需要操作数据库和文件
        System.out.println("导出数据" + data + "到数据库备份文件");
        return true;
    }
}

File: 品味Java的21種設計模式/dp_cpp/factorymethod/example1/ExportDB.cpp
#include "ExportDB.h"

namespace cn {
namespace javass {
namespace dp {
namespace factorymethod {
namespace example1 {

bool ExportDB::export(std::string data) {
    //简单示意一下，这里需要操作数据库和文件
    puts("导出数据" + data + "到数据库备份文件");
    return true;
}
}
}
}
}
}
